### PR TITLE
Conditionally build ros subpackage

### PR DESCRIPTION
**Pull request by subpop**
Only build the ros subpackage when rpmbuild is invoked with the "--with ros" flag.

### CODE REVIEW

File: insights-client.spec [hunk 1]
```diff
@@ -1,5 +1,9 @@
 %define _binaries_in_noarch_packages_terminate_build 0
 
+# This conditional build macro adds a "--with ros" commandline option to
+rpmbuild. The default behavior is to build without it.
```
Comment (ptoscano) [inline insights-client.spec:4]: missing comment character here, which breaks the build

**Review by ptoscano — CHANGES_REQUESTED** [review]
Other than the spec syntax issue, there's the "condtionally" typo in both the PR title and the commit message.

File: insights-client.spec [hunk 2]
```diff
@@ -54,6 +59,7 @@ Summary: The subpackage for Insights resource optimization service
 The ros subpackage add ros_collect configuration parameter to insights-client.conf file,
 the parameter is set to True by default. The system starts sending PCP archives to
 Resource Optimization service upon modifying ros_collect parameter to True.
+%{endif}
```
Comment (ptoscano) [inline insights-client.spec:62]:
```suggestion
%endif
```

File: insights-client.spec [hunk 3]
```diff
@@ -90,6 +97,7 @@ ros_collect=True
 ### End insights-client-ros ###
 EOF
 fi
+%{endif}
```
Comment (ptoscano) [inline insights-client.spec:100]:
```suggestion
%endif
```

File: insights-client.spec [hunk 4]
```diff
@@ -112,9 +120,10 @@ if [ $1 -eq 0 ]; then
     rm -f %{_sysconfdir}/insights-client/.*.etag
 fi
 
+%if %{with ros}
 %postun ros
 sed -i '/### Begin insights-client-ros ###/,/### End insights-client-ros ###/d;/ros_collect=True/d' %{_sysconfdir}/insights-client/insights-client.conf
-
+%{endif}
```
Comment (ptoscano) [inline insights-client.spec:126]:
```suggestion
%endif
```

**Comment by subpop** [comment]
I fixed the typos and added an empty `%files ros` section, which I discovered is required in order to actually create the RPM.

**Pull request merged**